core/src/main/scala/org/http4s/DecodeFailure.scala

```diff
@@ -0,0 +1,45 @@
+package org.http4s
+
+import scalaz.{\/-, -\/, Equal}
+
+trait DecodeFailure {
+  def msg: String
+}
+
+final case class DecodeFailureException(failure: DecodeFailure) extends RuntimeException(failure.msg)
```

Comment:
Maybe I'm just ignorant, but I can't figure out what this wrapper is for. My best guess if for situations where you need to use `throw catch` but I don't see that anywhere.